### ISSUE TITLE
requestIdleCallback not available in Safari

### ISSUE THREAD

**Issue by frontsideair** [remix-run/indie-stack#124]
### Have you experienced this bug with the latest version of the template?

Yes

### Steps to Reproduce

- Start a dev server
- Navigate to it with Safari

### Expected Behavior

It should mount React

### Actual Behavior

It doesn't mount React, get an error in console:

> ReferenceError: Can't find variable: requestIdleCallback 

Unfortunately `rIC` is not supported by Safari, according to [caniuse.com](https://caniuse.com/requestidlecallback). Either its existence should be checked or it should be polyfilled. (Or it can be removed altogether.) I can provide the PR depending on the decision.

**Comment by kentcdodds** [comment]
Ah dang. I had no idea 😬 Safari really is the new IE 🙃

Yeah, a PR would be welcome. Let's just do this:

```ts
const rIC = requestIdleCallback ?? setTimeout
rIC(() => {
  // ... etc.
})
```

**Comment by frontsideair** [comment]
Yeah, I had to double check to make sure it's not supported. 😂 I can create a PR, but checking the [polyfill](https://github.com/behnammodi/polyfill/blob/master/window.polyfill.js#L7-L24) linked in MDN, it seems like we should add a timeout of 1 milliseconds (see `relaxation`). The whole polyfill is not needed as we don't use the args. What do you think?

**Comment by kentcdodds** [comment]
Sounds good 👍
